node/src/accountant/mod.rs

```diff
@@ -2078,208 +2576,303 @@ mod tests {
 
     #[test]
     fn accountant_scans_after_startup() {
```

Comment:
This test is quite long. I know it would be hard to refactor it because the preparatory lines are later used in assertions. So, please give it a try in case you see something that can be refactored out.